packages/react/src/grouped-fields/docs/overview.mdx

```diff
@@ -0,0 +1,256 @@
+---
+title: Grouped fields
+description: The Grouped fields component allows two closely related form inputs to be grouped together.
+group: Forms
+storybookPath: /story/forms-groupedfields--basic
+unreleased: true
+figmaGalleryNodeId: REPLACE_ME
+relatedComponents:
+  [
+    'autocomplete',
+    'combobox',
+    'date-picker',
+    'password-input',
+    'search-input',
+    'select',
+    'text-input',
+    'textarea',
+  ]
+---
+
+```jsx live
+<GroupedFields legend="Item weight">
+	{({ field1Props, field2Props }) => (
+		<>
+			<TextInput label="Weight" {...field1Props} />
+			<Select
+				label="Unit"
+				options={[
+					{ value: 'mg', label: 'mg' },
+					{ value: 'g', label: 'g' },
+					{ value: 'kg', label: 'kg' },
+					{ value: 'ton', label: 'ton' },
+				]}
+				hideOptionalLabel
+				maxWidth="sm"
+				value="kg"
+				{...field2Props}
+			/>
+		</>
+	)}
+</GroupedFields>
+```
+
+Grouped fields allows two related form inputs to be composed together while providing the ability to present and describe this relationship and any invalid input errors.
+
+It uses the [render props pattern](https://www.patterns.dev/react/render-props-pattern/) to spread a series of necessary attributes to the `children` form inputs.
+
+<DoHeading />
+
+- group two related inputs, such as weight and unit, and date and time
+- always provide a meaningful legend, regardless if it’s visually hidden
```

Comment:
```suggestion
- always provide a meaningful legend, if it is visible or visually hidden
```

Reply:
I feel like we need "regardless" again:

- always provide a meaningful legend, regardless if it is visible or visually hidden

Reply:
agreed, please proceed